src/miniscript/context.rs

```diff
@@ -614,7 +614,7 @@ impl ScriptContext for Tap {
         // When the transaction sizes get close to block limits,
         // some guarantees are not easy to satisfy because of knapsack
         // constraints
-        if ms.ext.pk_cost > MAX_BLOCK_WEIGHT as usize {
+        if ms.ext.pk_cost as u64 > Weight::MAX_BLOCK.to_wu() {
```

Comment:
In e7b5db2cc27f1c6b79abf23cffd29d13469c9245:

In a followup we should change the units of `pk_cost` to be `Weight` so we don't need to cast.

Reply:
I added it to my todo list.